src/emails/getEmailL10n.node.ts

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { dirname, resolve } from "path";
+import { fileURLToPath } from "url";
+import { readFileSync, readdirSync } from "fs";
+import { FluentBundle, FluentResource } from "@fluent/bundle";
+import { MarkupParser, ReactLocalization } from "@fluent/react";
+import { Fragment, createElement } from "react";
+import { acceptedLanguages, negotiateLanguages } from "@fluent/langneg";
+import { JSDOM } from "jsdom";
+import type { ExtendedReactLocalization, GetFragment } from "../app/hooks/l10n";
+import type { SanitizedSubscriberRow } from "../app/functions/server/sanitize";
+
+export function getEmailL10n(
+  subscriber: SanitizedSubscriberRow,
+): ExtendedReactLocalization {
+  // We don't have a runtime language when we email people, so use their
+  // language setting from when they signed up for their Mozilla account:
+  return getL10n(getL10nBundles(subscriber.signup_language ?? "en"));
+}
+
+export type LocaleData = {
+  locale: string;
+  bundleSources: string[];
+};
+
+/**
+ * Get the localisation sources for the locales relevant to the current user
+ *
+ * This function can run on the server side, and only returns serialisable data.
+ * This means that it can either be used to construct a ReactLocalization object
+ * on the server side, or be passed to Client Component to construct such an
+ * object on the client side.
+ *
+ * @param userLocales The user's preferred locales, in the syntax of the Accept-Language HTTP header
+ * @returns The sources for l10n bundles that can be used to construct a ReactLocalization object
+ */
+export function getL10nBundles(userLocales: string): LocaleData[] {
+  const languages = userLocales ? acceptedLanguages(userLocales) : [];
+  const supportedLocales = process.env.SUPPORTED_LOCALES!.split(",");
+  // const availableLocales = Object.keys(bundleSources);
+  // const filteredLocales =
+  //   process.env.APP_ENV === "heroku"
+  //     ? availableLocales
+  //     : availableLocales.filter((locale) => supportedLocales?.includes(locale));
+  // const currentLocales = negotiateLanguages(languages, filteredLocales, {
```

Comment:
```suggestion

```

Reply:
I removed the whole file in https://github.com/mozilla/blurts-server/pull/4401, but might as well fix it for a cleaner history :) 49c09b4a2